src/main/java/com/maddyhome/idea/vim/extension/nerdtree/Mappings.kt

```diff
@@ -0,0 +1,159 @@
+    }
+  }
+
+  /**
+   * Registers navigation-related mappings
+   * <pre><code>
+   * Default~
+   * Key      Description                                                             Map settings
+   *
+   * O........Recursively open the selected directory..................*NERDTreeMapOpenRecursively*
+   * x........Close the current nodes parent..................................*NERDTreeMapCloseDir*
+   * X........Recursively close all children of the current node.........*NERDTreeMapCloseChildren*
+   *
+   * P........Jump to the root node...........................................*NERDTreeMapJumpRoot*
+   * p........Jump to current nodes parent..................................*NERDTreeMapJumpParent*
+   * K........Jump up inside directories at the current tree depth......*NERDTreeMapJumpFirstChild*
+   * J........Jump down inside directories at the current tree depth.....*NERDTreeMapJumpLastChild*
+   * <C-J>....Jump down to next sibling of the current directory.......*NERDTreeMapJumpNextSibling*
+   * <C-K>....Jump up to previous sibling of the current directory.....*NERDTreeMapJumpPrevSibling*
+   * </code></pre>
+   */
+  fun registerNavigationMappings() {
+    // TODO support going [count] lines upward/downward or to line [count]
+    register("k", Action.ij("Tree-selectPrevious"))
+    register("j", Action.ij("Tree-selectNext"))
+    register("G", Action.ij("Tree-selectLast"))
+    register("gg", Action.ij("Tree-selectFirst"))
+
+    // FIXME lazy loaded tree nodes are not expanded
+    register("NERDTreeMapOpenRecursively", "O", Action.ij("FullyExpandTreeNode"))
+    // This action respects `ide.tree.collapse.recursively`. We may prompt the user to disable it
+    register("NERDTreeMapCloseDir", "x", Action { _, tree ->
+      tree.selectionPath?.parentPath?.let {
+        if (tree.getRowForPath(it) >= 0) { // skip if invisible, but we cannot use `tree.isVisible(path)` here
+          tree.selectionPath = it
+          tree.collapsePath(it)
+          tree.scrollPathToVisible(it)
+        }
+      }
+    })
+    register(
+      "NERDTreeMapCloseChildren",
+      "X",
+      Action { _, tree ->
+        val path = tree.selectionPath ?: return@Action
+
+        // FIXME We should avoid relying on `ide.tree.collapse.recursively` since it closes visible paths only
+        val recursively = AdvancedSettings.getBoolean("ide.tree.collapse.recursively")
+        AdvancedSettings.setBoolean("ide.tree.collapse.recursively", true)
+        // Note that we cannot use `tree.collapsePaths` here since it does not respect `ide.tree.collapse.recursively`
+        TreeUtil.listChildren(path.lastPathComponent as TreeNode).filterNot(TreeNode::isLeaf)
+          .map(path::pathByAddingChild).forEach(tree::collapsePath)
+        AdvancedSettings.setBoolean("ide.tree.collapse.recursively", recursively)
+
+        tree.scrollPathToVisible(path)
+      },
+    )
+
+    register("NERDTreeMapJumpRoot", "P", Action.ij("Tree-selectFirst"))
```

Comment:
The first row is not the root of "External Libraries" in IntelliJ.
Is it supposed to jump to the root of project files instead of libraries?